src/object/obj_tx.c

```diff
@@ -332,6 +344,8 @@ dc_tx_check_pmv_internal(daos_handle_t th, struct dc_tx **ptx)
 	if (tx == NULL)
 		return -DER_NO_HDL;
 
+	D_MUTEX_LOCK(&tx->tx_lock);
```

Comment:
dc_tx_check_pmv is called from src/object/cli_obj.c.   Where does the lock get released in this case?

Reply:
dc_tx_check_pmv calls dc_tx_check_pmv_internal with a NULL ptx. The lock is, therefore, released... (See below.)